batch/batch/google_storage.py

```diff
@@ -85,6 +137,21 @@ def _read_binary_gs_file(self, uri, *args, **kwargs):
         content = f.download_as_string(*args, **kwargs)
         return content
 
+    def _read_gs_file_to_filename(self, uri, filename, *args, **kwargs):
+        bucket, path = GCS._parse_uri(uri)
+        bucket = self.gcs_client.bucket(bucket)
+        f = bucket.blob(path)
+        f.metadata = {'Cache-Control': 'no-cache'}
+        f.download_to_filename(filename, *args, **kwargs)
+
+    def _read_gs_file_to_file(self, uri, file, offset, *args, **kwargs):
+        file.seek(offset)
+        bucket, path = GCS._parse_uri(uri)
+        bucket = self.gcs_client.bucket(bucket)
+        f = bucket.blob(path)
+        f.metadata = {'Cache-Control': 'no-cache'}
+        f.download_to_file(file, *args, **kwargs)
+
     def _delete_gs_files(self, uri_prefix):
         bucket, prefix = GCS._parse_uri(uri_prefix)
         bucket = self.gcs_client.bucket(bucket)
```

Comment:
We probably should issue these delete requests in parallel. We can benchmark and decide later though.

Reply:
I'm not going to worry about this right now. I can add an asana task.